lib/OrdinaryDiffEqRosenbrock/src/rosenbrock_caches.jl

```diff
@@ -663,99 +697,23 @@ end
 
 ### Rodas4 methods
 
-struct Rodas4ConstantCache{TF, UF, Tab, JType, WType, F, AD} <: RosenbrockConstantCache
+struct Rodas4ConstantCache{TF, UF, Tab, JType, WType, F, AD, rateType} <: RosenbrockConstantCache
     tf::TF
     uf::UF
     tab::Tab
     J::JType
     W::WType
     linsolve::F
     autodiff::AD
+    ks::Vector{rateType}
```

Comment:
why would this cache?

Reply:
this was cached because otherwise we needed to make this vector new at each step. (since before we had separate variables `k1`, `k2` etc but we needed them to exist in an indexable collection.